src/Domain/SubscriptionRepository.php

```diff
@@ -11,6 +11,14 @@
  */
 interface SubscriptionRepository {
 
+	/**
+	 * Add or update a subscription.
+	 *
+	 * If the subscription exists in the repository, update it.
+	 *
+	 * @param Subscription $subscription
+	 * @return mixed
```

Comment:
That means the repositories should catch the implementation-specific exceptions and throw their own exceptions, right? Shall I do a PR that does that?